### PR TITLE
Add modal window for transferring data between computers

### PR DESCRIPTION
**Pull request by JamieDavidson**
Inspired by #28

Adds a modal window allowing for a token to be copied between computers, restoring the current state on target machine.

Pictures
Modal window
image

The copy button has a tooltip
image

And the modal validates strings that aren't base64...
image

...as well as base64 strings that aren't JSON payloads
image

The token is comprised of a base64 encoded JSON object containing the key value pairs from application storage.
For example:
```
eyJyczNkYWlseS11cGRhdGVkIjoiMTcxNDMyMDU0Nzc3MiIsInJzM2RhaWx5c2hvcHMtdXBkYXRlZCI6IjE3MTQzMjA1ODQ5ODQiLCJtZW5hcGhvcy1vYmVsaXNrIjoidHJ1ZSIsInRyZWFzdXJlLWh1bnRlci1rZXlzIjoidHJ1ZSIsInNlc2lvbl90b2tlbl91c2VyIjoibnVsbCIsInNob290aW5nLXN0YXIiOiJ0cnVlIiwic2VyZW5pdHktcG9zdCI6InRydWUiLCJ0cmF2ZWxpbmctbWVyY2hhbnQiOiJ0cnVlIiwiY2hha3JhLXVpLWNvbG9yLW1vZGUiOiJsaWdodCIsIkFNUF91bnNlbnRfMDY3YjliMDdjOCI6IltdIiwiZ29yYWpvLWNhcmQiOiJ0cnVlIiwiZGFpbHktY2hhbGxlbmdlIjoidHJ1ZSIsInJ1bmVzcGhlcmUiOiJ0cnVlIn0=
```

Is equivalent to
```
{"rs3daily-updated":"1714320547772","rs3dailyshops-updated":"1714320584984","menaphos-obelisk":"true","treasure-hunter-keys":"true","sesion_token_user":"null","shooting-star":"true","serenity-post":"true","traveling-merchant":"true","chakra-ui-color-mode":"light","AMP_unsent_067b9b07c8":"[]","gorajo-card":"true","daily-challenge":"true","runesphere":"true"}
```

I tried to pick a suitable colour palette based on the existing colour scheme. Please let me know if you have suggestions for different colours.

I've also changed a number of paths that failed to load locally, these were mostly the rsdata paths as these are in a separate repository. I found a mix of fully qualified URLs and relative paths, and where applicable I've changed these to be the fully qualified URL `dailyscape.github.io/rsdata`. Please do let me know if you'd rather I undo these changes.

### CODE REVIEW

**Comment by 607ch00** [comment]
Cool contribution! Thank you.  

Would you be able to remove the code styling and changing the paths on resources from this PR please? It would be easier for me to review and get it merged in quicker if the changes are isolated to just the specific goal here.

I'm open to discussing those but as a separate PR.  One, I saw you had changed some opengraph URLs which need to be a full URL so services can look up the thumb etc, but reverted, thank you.  For the relative paths, I have it set up this way for local testing if I need it (use symlinks to the other repo(s) being referenced)

**Comment by JamieDavidson** [comment]
Sorry about that! That was mostly a very opinionated IDE's auto format (and one that isn't HTML friendly). I've removed the formatting changes, it should be much easier to review now :)

File: index.html [hunk 1]
```diff
@@ -109,6 +109,9 @@ <h6 class="dropdown-header">RS3 Discords</h6>
                 <li class="nav-item">
                     <a class="nav-link expanding_button" href="#" id="layout-button" title="Toggle Layout">⊟ <span class="expanding_text"> Compact Mode</span></a>
                 </li>
+                <li class="nav-item">
+                    <a class="nav-link expanding_button" href="#" id="token-button" title="Import / Export" data-bs-toggle="modal" data-bs-target="#token-modal">🖫 <span class="expanding_text">&nbsp;Import / Export</span></a>
```
Comment (607ch00) [inline index.html:113]:
```suggestion
                    <a class="nav-link expanding_button" href="#" id="token-button" title="Import / Export" data-bs-toggle="modal" data-bs-target="#token-modal">🖪 <span class="expanding_text">&nbsp;Import / Export</span></a>
```
Does this icon work on your system? The other one isn't working for me

**Comment by 607ch00** [comment]
Cool stuff, it seems to work well here. I'm going to go for it and add that icon instead

**Pull request merged**